File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k=k%len(nums)
        def rev(nums,start,end):
            while start<=end:
                nums[start],nums[end]=nums[end],nums[start]
                start+=1
                end-=1
        rev(nums,0,(len(nums)-k)-1)
        rev(nums,len(nums)-k,len(nums)-1)
        rev(nums,0,len(nums)-1)
        return nums

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public static void ra(int[] nums,int s,int e){
        while(s<e){
            int t=nums[s];
            nums[s]=nums[e];
            nums[e]=t;
            s++;
            e--;
        }

    }
    public void rotate(int[] nums, int k) {
        int n=nums.length;
         k=k%n;
        if(k==0||n==0){
            return ;
        }
        ra(nums,0,n-1);
        ra(nums,0,k-1);
        ra(nums,k,n-1);
    }
}
